lib-impl/src/main/java/org/ergoplatform/appkit/impl/BlockchainContextImpl.java

```diff
@@ -98,7 +98,14 @@ ApiClient getApiClient() {
         for (OutputInfo box : boxes) {
             String boxId = box.getBoxId();
             ErgoTransactionOutput boxInfo = ErgoNodeFacade.getBoxById(_retrofit, boxId);
-            returnList.add(new InputBoxImpl(this, boxInfo));
+            // can be null if node does not know about the box (yet)
+            // instead of throwing an error, we continue with the boxes actually known
+            if (boxInfo != null) {
+                returnList.add(new InputBoxImpl(this, boxInfo));
+            } else {
+                // add a placeholder - otherwise we cannot determine if we are at the end of the list
+                returnList.add(null);
```

Comment:
I don't think this is a good idea. It is better to filter such boxes out. 
In order to detect the last chunk the `chunk.size() < DEFAULT_LIMIT_FOR_API` can be replaced with `chunk.size() == 0`.
This will require one additional request, but on the other hand will not rely on each chuck be of the DEFAULT_LIMIT_FOR_API size.